nessie/s3_gateway_test.go

```diff
@@ -0,0 +1,107 @@
+package nessie
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math/rand"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/spf13/viper"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+var sigs = []struct {
+	Name           string
+	GetCredentials func(id, secret, token string) *credentials.Credentials
+}{
+	{"V2", credentials.NewStaticV2},
+	{"V4", credentials.NewStaticV4},
+}
+
+const (
+	numUploads           = 100
+	randomDataPathLength = 1020
+	prefix               = "main/data/"
+)
+
+func TestS3UploadAndDownload(t *testing.T) {
+	const parallelism = 10
+
+	ctx, _, repo := setupTest(t)
+
+	accessKeyID := viper.GetString("access_key_id")
+	secretAccessKey := viper.GetString("secret_access_key")
+	endpoint := viper.GetString("s3_endpoint")
+	opts := minio.PutObjectOptions{}
+
+	for _, sig := range sigs {
+		t.Run("Sig"+sig.Name, func(t *testing.T) {
+			// Use same sequence of pathnames to test each sig.
+			rand := rand.New(rand.NewSource(17))
+
+			creds := sig.GetCredentials(accessKeyID, secretAccessKey, "")
+
+			type Object struct {
+				Path, Content string
+			}
+
+			var (
+				wg      sync.WaitGroup
+				objects = make(chan Object, parallelism*2)
+			)
+
+			for i := 0; i < parallelism; i++ {
+				wg.Add(1)
+				go func() {
+					client, err := minio.New(endpoint, &minio.Options{
+						Creds:  creds,
+						Secure: false,
+					})
+					if err != nil {
+						t.Error(fmt.Sprintf("minio.New: %s", err))
```

Comment:
why aren't you returning in case of error? will be hard to continue without a client

Reply:
Good catch!  Dying fatally outside the goroutine when this fails.